### PR TITLE
[MM-25673] Stabilize master

### PR DESCRIPTION
**Pull request by ethervoid**
#### Summary

Multiple changes to stabilize gorp:

- Fix tests
- Partially revert #3 to avoid breaking the `Rename` and some `Select` function cases
- Include CircleCI configuration
- Make Ginkgo work

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-25673

### CODE REVIEW

File: gorp.go [hunk 1]
```diff
@@ -257,15 +257,15 @@ func columnToFieldIndex(m *DbMap, t reflect.Type, name string, cols []string) ([
 			cArguments := strings.Split(field.Tag.Get("db"), ",")
 			fieldName = cArguments[0]
 
+			if fieldName == "" || fieldName == "-" {
```
Comment (lieut-data) [inline gorp.go:260]: With respect to the partial revert, can you elaborate on the impact to the bot fields in question?

**Review by lieut-data — COMMENTED** [review]
One comment, but shifting review to @crspeller given the impact to his original changes.

File: gorp_test.go [hunk 2]
```diff
@@ -48,9 +48,10 @@ var (
 	debug bool
 )
 
-func init() {
+func TestMain(m *testing.M) {
```
Comment (agnivade) [inline gorp_test.go:53]: :+1:

File: go.mod [hunk 3]
```diff
@@ -3,11 +3,11 @@ module github.com/mattermost/gorp
 go 1.13
 
 require (
-	github.com/go-gorp/gorp v2.2.0+incompatible
```
Comment (agnivade) [inline go.mod:6]: :+1:

File: gorp.go (hunk 1 above)
Comment (agnivade) [inline gorp.go:260]: Just wanted to clarify quickly: is this change already covered by the change in the unit tests? If not, can we add some tests to cover them?

**Review by agnivade — COMMENTED** [review]
Really nice work! Just some small comments.

**Comment by agnivade** [comment]
Requesting a CI review from Elisabeth.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+          MYSQL_DATABASE: gorptest
+          MYSQL_USER: gorptest
+          MYSQL_PASSWORD: gorptest
+    steps:
+      - checkout
+      - run:
+          name: Run Tests
+          command: |
+            go get -u github.com/onsi/ginkgo/ginkgo
+            ./test_all.sh
```
Comment (metanerd) [inline .circleci/config.yml]: In this bash script there are also remnants of travis and our blacklisting does not recognize the name `test_all.sh`. Personally I would favour converting it into a Makefile to go with the standard, but we can also do this at a later iteration.

**Review by metanerd — CHANGES_REQUESTED** [review]
Thanks for taking care!

File: gorp.go (hunk 1 above)
Comment (ethervoid) [reply gorp.go:260]: Yes it is covered but I’ll add more explicit tests to be really sure

File: .circleci/config.yml (hunk 4 above)
Comment (ethervoid) [reply .circleci/config.yml]: Ok, I’ll tackle it

File: gorp.go (hunk 1 above)
Comment (ethervoid) [reply gorp.go:260]: Should be none and we recover the Rename feature

Comment (crspeller) [reply gorp.go:260]: This should be fine. We were not using the rename feature at the time so I didn't worry about breaking it. It keeps the change acutally needed from the other PR.

**Review by crspeller — APPROVED** [review]

File: .circleci/config.yml (hunk 4 above)
Comment (ethervoid) [reply .circleci/config.yml]: Done

**Review by agnivade — APPROVED** [review]
Fantastic

**Review by metanerd — APPROVED** [review]
Great! :+1:

**Pull request merged**